### DISCUSSION TITLE
ImageMagick 7.0.10-43 -auto-orient Fails to Recognize Option

### DISCUSSION THREAD

**Discussion by DCastelino** [ImageMagick/ImageMagick#3230] [Development]
I'm using image magick -auto-orient in javascript, creating an ActiveXObject, and using -auto-orient to rotate images uploaded through iPhones. It's not working though! I keep getting the error mogrify: 410: unrecognized option '-auto-orient'.

Anyone experience anything like this?

**Comment by urban-warrior** [comment]
Type `mogrify -help | grep auto-orient`.  Is `-auto-orient` listed as an option?  Type `mogrify --version`.  What release of ImageMagick are you using?  Perhaps you have an older version of ImageMagick that must be upgraded.

**Reply by DCastelino** [reply]
Awesome! THank you @urban-warrior for the reply! It turns out it looks like the HDRI verison of imagemagick doesn't seem to work with -auto-orient. I reinstalled the regular version of imagemagick, and now it works! The commands you provided were really helpful in debugging!

Thank you so much!

**Reply by fmw42** [reply]
Works fine for me on IM 7.0.11.0 Q16 Mac OSX

`magick mogrify -help | grep auto-orient`
  -auto-orient         automagically orient (rotate) image

**Reply by aruprakshit** [reply]
My version is:

```
 magick -version
Version: ImageMagick 7.1.0-1 Q16 x86_64 2021-06-21 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib
```

But I see errrors:

```
Data error: File Failed to manipulate with MiniMagick, maybe it is not an image? Original Error: `mogrify -auto-orient! /var/folders/r8/4mc1_kfx3qq704csrjdy9c8m0000gn/T/mini_magick20230302-2275-1u6m8ia.jpg` failed with error:
mogrify: unrecognized option `-auto-orient!' @ error/mogrify.c/MogrifyImageCommand/4250.
```

**Reply by snibgo** [reply]
IM has no option "`-auto-orient!`", with the bang (exclamation mark).

**Comment by snibgo** [comment]
As far as I know, "-auto-orient" works fine, whether the build is HDRI or not.

@DCastelino: please tell us what version "-auto-orient" does not seem to work. What is the full text output from "magick -version"?

**Reply by DCastelino** [reply]
The ImageMagick version was in version 7.0.10-46 Q16 HDRI (x86)

I kept getting the unrecognized command -auto-orient in the error logs. Will post the full text output on this thread tomorrow.

**Reply by DCastelino** [reply]
Here was the result of running the command magick -version. I had it load balanced between two servers, and it was installed on only one of the machines. I suspect that was the problem. I installed ImageMagick on the other machine, and everything seems to work okay now. Thank you all for the help!

Version: ImageMagick 7.0.10-43 Q16 x86 2020-11-23 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 192829333
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib
